### PR TITLE
meta: Eslint JSDoc plugin

### PR DESCRIPTION
**Pull request by canerakdas**
## Description

I thought the Eslint JSDoc rules would make sense for comments that we might miss or were not properly formatted. I've updated the missing or incorrectly indented sections with this PR

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/api-docs-tooling/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: eslint.config.mjs
Comment (RedYetiDev) [inline eslint.config.mjs:1]: Is there a reason why the recommended config can't be used?

jsdoc.configs['flat/recommended-error']

Comment (canerakdas) [reply eslint.config.mjs:1]: There seem to be stricter rules than the ones we apply, I can even say that I am against some of the rules being applied as recommended

I tried to combine the rules we already apply as much as possible when using/developing features or for a more efficient review process

**Review by AugustinMauroy — APPROVED** [review]
LGTM !

**Review by ovflowd — APPROVED** [review]

**Comment by ovflowd** [comment]
Thanks for the work here <3

**Pull request merged**